core/src/replicator.rs

```diff
@@ -232,30 +308,61 @@ impl Replicator {
         }
     }
 
-    fn wait_for_ledger_download(&self) {
+    fn wait_for_ledger_download(
+        start_slot: u64,
+        blocktree: &Arc<Blocktree>,
+        exit: &Arc<AtomicBool>,
+        node_info: &ContactInfo,
+        cluster_info: &Arc<RwLock<ClusterInfo>>,
+    ) {
         info!("window created, waiting for ledger download done");
         let _start = Instant::now();
         let mut _received_so_far = 0;
 
+        let mut current_slot = start_slot;
+        let mut done = false;
         loop {
-            if let Ok(entries) = self.blocktree.get_slot_entries(self.slot, 0, None) {
-                if !entries.is_empty() {
+            loop {
```

Comment:
This double loop is not easy to read :S 

Does this do the same thing? 

```
        'outer: loop {
            while let Ok(meta) = blocktree.meta(current_slot) {
                if let Some(meta) = meta {
                    if meta.is_rooted {
                        current_slot += 1;
                        info!("current slot: {}", current_slot);
                        if current_slot >= start_slot + ENTRIES_PER_SEGMENT {
                            info!("current slot: {} start: {}", current_slot, start_slot);
                            break 'outer;
                        }
                        continue;
                    }
                }
                break;
            }
            if exit.load(Ordering::Relaxed) {
                break;
            }
            sleep(Duration::from_secs(1));
        }
```

Reply:
yea, that's better.